src/NHibernate/Type/EntityType.cs

```diff
@@ -441,10 +442,7 @@ public string RHSUniqueKeyPropertyName
 			get { return uniqueKeyPropertyName; }
 		}
 
-		public virtual string PropertyName
-		{
-			get { return null; }
-		}
+		public virtual string PropertyName { get; set; }
```

Comment:
This looks hacky. If property name is really required to be set in runtime - I think you should introduce instead some set method in `ManyToOne`

But it seems you should be able to set it similair way it's set for `OneToOneType`. Via the following chain: `HbmManyToOne.Name` -> `ManyToOne.PropertyName` (add property) -> `ManyToOneType.PropertyName` (add ctor parameter)

See `PropertiesBinder.BindManyToOne`  for a starting point.